### PR TITLE
Thuang 650 my collection page feedback

### LINKED ISSUE #650: My Collections Page
Context:

This ticket implements the My Collections Page.

Figma: https://www.figma.com/file/g0L0CfVzRc8r6HQa4NH1wJ/Publisher-Collection?node-id=36%3A1457

NOTE: Please use to the latest design [here](https://www.figma.com/file/g0L0CfVzRc8r6HQa4NH1wJ/Collections?node-id=0%3A1) and sync with @harley on which Blueprint components to use

What:

NOTE (11/6/2020): The API below may have changed, please sync with BE peeps for the right endpoints to use! Thank you!

1. Route: `/my-collections`

1. Use `GET /collections` to get all the collection ids that belong to this user

1. Use `GET /collections/{id}` to retrieve each collection's data

1. Use directory `/views/MyCollections` to construct the page

1. Use React Query to build queries where necessary

**TEAM NOTES**
1. Part 2 ticket: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/737

### PR DESCRIPTION
**Pull request by tihuan**
demo

(12/2) Seve added New Tag:
Screen Shot 2020-12-02 at 10 12 32 AM

@hthomas-czi PTAL if the UI looks good! Thanks a lot!

resolves #650

### CODE REVIEW

**Review by hthomas-czi — COMMENTED** [review]
Looks good, @seve!

I was originally thinking the Status and Assay overflow indicators would be `minimal` [Tags](https://blueprintjs.com/docs/#core/components/tag). Since it's not a button, I figured the popover would appear on hover but that's not a huge deal.

**Comment by tihuan** [comment]
Thanks so much for the quick feedback, @hthomas-czi 🙏 

I changed it from minimal to outlined, because of the same aligning issue we had with Create Collection component too 😆 But if the same solution doesn't work here, we can certainly do it differently!

@seve can you help update the button to show the Popover on hover please? And I think I also accidentally removed the window bound thing, so please help add it back too!

I'll start adding notes in the PR description and code, so you know what changes I made

After that, please take over the PR and work with Harley on design changes, so we can get this merged today! Thank you!

File: frontend/.eslintrc.js
Comment (tihuan) [inline frontend/.eslintrc.js:39]: I think I'm turning off the camelCase lint rule, since we just have so many of them from API responses 😆

File: frontend/src/common/entities.ts
Comment (tihuan) [inline frontend/src/common/entities.ts:68]: Add Collection Dataset's metadata type

File: frontend/src/common/queries/collections.ts
Comment (tihuan) [inline frontend/src/common/queries/collections.ts:17]: use `VISIBILITY_TYPE` and remove `VISIBILITY`

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/index.tsx:1]: Extract Popover for code splitting

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/style.ts
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/style.ts:11]: Harley wants us to use `PT_GRID_SIZE_PX` as base for styling as much as possible. Such as margin, padding, etc.

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/common/utils.ts
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/common/utils.ts:1]: This file does a collection's metadata aggregation

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx:25]: Code split Popover

**Comment by hthomas-czi** [comment]
> I changed it from minimal to outlined, because of the same aligning issue we had with Create Collection component too 😆 But if the same solution doesn't work here, we can certainly do it differently!

Good thinking! However, the aligning issue shouldn't be a problem here because the overflow indicator is a tag, not a button. Let me know if you have any questions about that.

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx:50]: This is commented out for now to show what it looks like to have both public and private collections in My Collections page, because a user can have `WRITE` access to both public and private collections that they own. But right now there's no such user in our database!

Before merging, we need to restore this back to:

```tsx
// If there is an explicity accessType only show collections with that accessType
if (props.accessType && collection.access_type !== props.accessType) {
  return null;
}
```

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/index.tsx:46]: since a user can have write access to both public and private collections, we need to use `collection.visibility` to fetch the data

File: frontend/src/components/Header/index.tsx
Comment (tihuan) [inline frontend/src/components/Header/index.tsx:15]: We actually need to conditionally render this with feature flag 🙏 Please help correct this! Thank you!

e.g., 

```tsx
import { get } from "src/common/featureFlags";
import { FEATURES } from "src/common/featureFlags/features";
import { BOOLEAN } from "src/common/localStorage/set";

{get(FEATURES.CREATE_COLLECTION) !== BOOLEAN.TRUE ? null : BUTTON_HERE }
```

File: frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/index.tsx
Comment (tihuan) [inline frontend/src/components/Collections/components/CollectionsGrid/components/CollectionRow/components/Popover/index.tsx:8]: To avoid circular dependency such as a parent importing a child, and the child imports something from the parent, I split out common styles in `/common/style`. And this way it's more explicit what styles are shared, so when we update code in common styles, we are naturally aware of its potential side effects

**Comment by tihuan** [comment]
> Looks good, @seve!
> 
> I was originally thinking the Status and Assay overflow indicators would be `minimal` [Tags](https://blueprintjs.com/docs/#core/components/tag). Since it's not a button, I figured the popover would appear on hover but that's not a huge deal.

Ooh got it! So we should use https://blueprintjs.com/docs/#core/components/tag instead 👍 Thank you!

@seve can you help update it to use tag and overwrite the default Blueprint style following @hthomas-czi 's style guide in Figma [here](https://www.figma.com/file/JsWBZz7BuKhNcFqzYNJZLO/Design-System?node-id=444%3A0)? Thanks a lot!

Screen Shot 2020-11-30 at 10 38 56 AM

**Comment by hthomas-czi** [comment]
You've got it! Thanks, Timmy.

File: frontend/src/components/Collections/components/CollectionsGrid/CollectionRow/index.tsx
Comment (seve) [inline frontend/src/components/Collections/components/CollectionsGrid/CollectionRow/index.tsx]: Visibility was added as a prop so the grid components could be used for the main collections view as well

File: frontend/src/global.scss
Comment (tihuan) [inline frontend/src/global.scss:3]: REAL COOL! Thanks so much for adding this, @seve !!

**Comment by tihuan** [comment]
LGTM! I also added a screenshot in the description, so it's easier for @hthomas-czi to review 😄 👌 Thanks, @seve !!

**Review by hthomas-czi — APPROVED** [review]
Beautiful! Great work!

**Pull request merged**
